tests/arch/arm/arm_interrupt/testcase.yaml

```diff
@@ -1,19 +1,14 @@
+common:
+  filter: CONFIG_ARMV6_M_ARMV8_M_BASELINE or CONFIG_ARMV7_M_ARMV8_M_MAINLINE
+  tags: arm interrupt ignore_faults
+  arch_allow: arm
```

Comment:
isn't this redundant to the filter?

Reply:
Not sure (@nashif to confirm)
I think the arch allow (earlier: whitelist) is required for proper filtering application

Reply:
All tests/arch/arm have the arch_allow: arm listed.

This PR does not change that. It is only grouping together common properties, in the test variants, and adds the !TRUSTED_EXECUTION_NONSECURE filter when necessary.

Reply:
I understand you're just refactoring. Seems worth cleaning up the redundancy while we're here though. @nashif, do we need the arch_allow when we're filtering on arm-specific kconfigs?

Reply:
@MaureenHelm, cannot make this work without the arch_allow directive; must be something with the sanity check filtering algorithm that prevents from successfully filtering-out platforms.

Reply:
ok, we can leave it as-is then